src/components/metadata/styles/metadata-code.scss

```diff
@@ -0,0 +1,178 @@
+
+  &--visible {
+    transform: translateX(0);
+  }
+
+  @mixin breakpoints($sidebar-width) {
+    // When screen is the minimum usable width
+    @media (min-width: $metadata-sidebar-code-breakpoint) {
+      // Anchor code to the left sidebar
+      left: $sidebar-width + $margin;
+    }
+
+    // When screen is wide enough to fit all sidebars
+    $sidebar-total-width: $sidebar-width + $margin + $metadata-sidebar-code-open +
+      $metadata-sidebar-width-open;
+    @media (min-width: #{$sidebar-total-width}) {
+      // Anchor to right sidebar
+      left: calc(
+        100% - #{$margin + $metadata-sidebar-code-open +
+          $metadata-sidebar-width-open}
+      );
+    }
+  }
+
+  // When sidebar is open
+  &--sidebarVisible {
+    @include breakpoints($sidebar-width-open);
+  }
+
+  // When sidebar is closed
+  &--no-sidebarVisible {
+    @include breakpoints($sidebar-width-closed);
+  }
+
+  @media print {
+    display: none;
+  }
+}
+
+.pipeline-metadata-code__title {
+  flex-grow: 1;
+  margin: 0;
+  margin: 2.1em 36px 1.8em 36px;
+  font-weight: normal;
+  font-size: 1.7em;
+  line-height: 1.6;
+}
+
+.pipeline-metadata-code__code {
+  position: absolute;
+  top: 7.1em;
+  right: $metadata-sidebar-width-open;
+  bottom: 0;
+  left: 0;
+  display: block;
+  padding: 0 0 0 36px;
+  overflow: auto;
+  color: var(--color-metadata-code-text);
+  font-size: 12.5px;
+  line-height: 1.8;
+  opacity: 1;
+  transition: opacity 0.4s ease 0.4s;
+
+  .pipeline-metadata-code--no-visible & {
+    opacity: 0;
+    transition: opacity 0.3s ease;
+  }
+
+  & * {
+    font-family: Consolas, Monaco, 'Courier New', Courier, monospace;
+  }
+
+  .hljs-keyword {
+    color: var(--color-metadata-code-text);
+  }
+}
+
+.pipeline-metadata-code pre {
+  width: 100%;
+  margin: 0;
+}
+
+.pipeline-metadata__code-toggle {
+  transition: opacity 0.2s ease;
+
+  &--no-hasCode {
+    opacity: 0.35;
+    pointer-events: none;
+  }
+}
+
+.pipeline-metadata__code-toggle {
+  margin-top: 0.3em;
+  margin-bottom: 2em;
+}
+
+.pipeline-metadata__code-toggle-input {
+  display: none;
+}
+
+.pipeline-metadata__code-toggle-label {
+  position: relative;
+  padding: 0 0 0 2.8em;
+  cursor: pointer;
+  user-select: none;
+
+  &:before,
+  &:after {
+    position: absolute;
+    top: 0.81em;
+    left: 0;
+    display: block;
+    margin: 0 0.8em 0 0;
+    background: $color-bg-light-1;
+    transform: translateY(-50%);
+    content: ' ';
+  }
+
+  &:before {
+    width: 1.7em;
+    height: 0.8em;
+    border-radius: 0.5em;
+    opacity: 0.3;
+    transition: opacity 0.15s ease;
+  }
+
+  &:after {
+    width: 0.8em;
+    height: 0.8em;
+    border-radius: 50%;
+    opacity: 0.7;
+    transition: opacity 0.15s ease, transform 0.15s ease;
+  }
+
+  &:hover:after {
+    opacity: 1;
+  }
+
+  &--checked:before {
+    background: $color-primary;
+  }
+
+  &--checked:after {
+    background: $color-primary;
+    transform: translate(0.9em, -50%);
```

Comment:
Reckon it might be better to use `%` rather than `em` for horizontal translation?